### ISSUE TITLE
[css-animations-2] interaction between the `timeline` property and `animation-timeline` property

### ISSUE THREAD

**Issue by graouts** [w3c/csswg-drafts#11054]
CSS Animations Level 2 has a [section](https://drafts.csswg.org/css-animations-2/#animations) detailing how setting properties via the Web Animations APIs can override related CSS properties. For instance:

> After a successful call to [play()](https://drafts.csswg.org/web-animations-1/#dom-animation-play) or [pause()](https://drafts.csswg.org/web-animations-1/#dom-animation-pause) on a [CSSAnimation](https://drafts.csswg.org/css-animations-2/#cssanimation), any subsequent change to the [animation-play-state](https://drafts.csswg.org/css-animations-1/#propdef-animation-play-state) will no longer cause the [CSSAnimation](https://drafts.csswg.org/css-animations-2/#cssanimation) to be played or paused as defined in [§ 4.5 The animation-play-state property](https://drafts.csswg.org/css-animations-2/#animation-play-state).

This section makes no mention of setting the `timeline` property of an animation. I believe the spec should detail what happens when setting `animation-timeline` after the `timeline` property has been modified via the JS bindings.

Additionally, should setting the `effect` property have any effect on further changes to the `animation-timeline` property? Currently, the spec says so:

> After successfully setting the [effect](https://drafts.csswg.org/web-animations-1/#dom-animation-effect) of a [CSSAnimation](https://drafts.csswg.org/css-animations-2/#cssanimation) to null or some [AnimationEffect](https://drafts.csswg.org/web-animations-1/#animationeffect) other than the original [KeyframeEffect](https://drafts.csswg.org/web-animations-1/#keyframeeffect), all subsequent changes to animation properties other than [animation-name](https://drafts.csswg.org/css-animations-1/#propdef-animation-name) or [animation-play-state](https://drafts.csswg.org/css-animations-1/#propdef-animation-play-state) will not be reflected in that animation. Similarly, any change to matching [@keyframes](https://drafts.csswg.org/css-animations-1/#at-ruledef-keyframes) rules will not be reflected in that animation. However, if the last matching `@keyframes` rule is removed the animation must still be canceled.

But the timeline being a property of the animation, and not the effect, should `animation-timeline` be excluded here?

**Comment by graouts** [comment]
Cc @birtles, @flackr, @andruud and @kevers-google.

**Comment by ydaniv** [comment]
+1 to both

**Comment by flackr** [comment]
Thanks for calling this out, this was an oversight. I agree with both choices.

**Comment by ydaniv** [comment]
@flackr can we resolve on this one async?

**Comment by ydaniv** [comment]
@graouts was this done via #11812 ?

**Comment by graouts** [comment]
> [@graouts](https://github.com/graouts) was this done via [#11812](https://github.com/w3c/csswg-drafts/pull/11812) ?

I believe that is correct! If @birtles or @flackr agree as well, I believe we can close this.

**Comment by flackr** [comment]
Agreed, I don't think we need a resolution, this falls within the realm of behavior that was always intended, but overlooked in the spec text.
